src/robot_state_publisher.cpp

```diff
@@ -91,8 +91,20 @@ std::string stripSlash(const std::string & in)
   return in;
 }
 
+std::string resolve(const std::string &prefix, const std::string &frame)
+{
+  if (prefix.empty())
+    return stripSlash(frame);
+
+  std::string composite;
+  composite.reserve(prefix.length() + frame.length());
+  composite.append(stripSlash(prefix));
+  composite.append(stripSlash(frame));
```

Comment:
I had to add `composite.append("/");` between L101 and L102: If we have tf_prefix "prefix" and frame "frame", with this version of "resolve", we'd have a frame published as "prefixframe" not "prefix/frame". Is this intended? If so, it would still be changed behaviour I believe?

I've tested this with a local version of @awinkler's xpp (where we are adding a workaround for this issue)

Reply:
I realise this relates to the discussion between @rhaschke and @sloretz above in the thread. As thus, 
(1) Is the consensus that all slashes - not only leading - should be stripped?
(2) Do we want to preserve the connecting-slash e.g. to work with the RobotModel plug-in in RViz?
(3) What happens if the prefix does not have a trailing slash (e.g. through prior user code specifying the prefix without a trailing slash)?